src/data/history.ts

```diff
@@ -419,17 +419,41 @@ const BLANK_UNIT = " ";
 export const computeHistory = (
   hass: HomeAssistant,
   stateHistory: HistoryStates,
+  entityIds: string[],
   localize: LocalizeFunc,
   sensorNumericalDeviceClasses: string[],
   splitDeviceClasses = false
 ): HistoryResult => {
   const lineChartDevices: { [unit: string]: HistoryStates } = {};
   const timelineDevices: TimelineEntity[] = [];
-  if (!stateHistory) {
+
+  const localStateHistory: HistoryStates = {};
+
+  // Create a limited history from stateObj if entity has no recorded history.
+  entityIds.forEach((entity) => {
+    if (entity in stateHistory) {
+      localStateHistory[entity] = stateHistory[entity];
+    } else if (hass.states[entity]) {
+      localStateHistory[entity] = [
+        {
+          s: hass.states[entity].state,
+          a: hass.states[entity].attributes,
+          lu: new Date(hass.states[entity].last_updated).getTime() / 1000,
+        },
+      ];
+    }
+  });
+  Object.keys(stateHistory).forEach((entityId) => {
```

Comment:
Can we combine `Object.keys(stateHistory)` and `entityIds` at the top so we dont have to go over `stateHistory` again?